text/0119-dynamic-component-creation.md

```diff
@@ -0,0 +1,335 @@
+
+This means that when the value provided to `lwc:is` changes the component and all of its children will be destroyed and recreated using the new constructor.  In addition, when the value provided to `lwc:is` is evaluated, the LWC engine will render depending on the following scenarios:
+
+- **The constructor is `undefined` or `null`**
+    - When the constructor is undefined the `<lwc.dynamic>` tag along with all of its children are not rendered to the DOM.
+- **The constructor is defined and is not a `LightningElementConstructor`**
+    - When the value provided to `lwc:is` is not a `LightningElementConstructor` then an error is thrown indicating the constructor value must be `LightningElementConstructor`.
+- **The constructor is defined and is a `LightningElementConstructor`**
+    - When the constructor is a `LightningElementConstructor`, the component will be rendered in place of `<lwc.dynamic>`. The tag name used for the dynamic component is the value of `lightningElementName` on `LightningElementConstructor`.
+
+_Note in the case of lazy loading, the component author is responsible for resolving the promise.  The only value that can be given to `lwc:is` is a `LightningElementConstructor`._
+
+#### Selecting the dynamic component
+
+It is important to note that `<lwc.dynamic>` will not be rendered to the DOM.  As such, it is not available to be selected using `this.template.querySelector`.  The actual custom element must be selected once it has been rendered to the DOM.  The dynamic component can be selected either through the `lightningElementName` property on the constructor or through another attribute assigned to the component.
+
+Component authors can use the `connectedCallback` of the dynamic component to signal when the component has been rendered to the DOM.
+
+#### Assigning attributes and event listeners
+
+Attributes and event listeners can be passed to the `<lwc.dynamic>` tag declaratively or can be assigned directly to the custom element imperatively.
+
+Attributes and event listeners assigned declaratively will be assigned to the dynamic component once it has been created.
+
+Declaratively:
+
+```html
+<template>
+    <lwc.dynamic lwc:is={ctor} class="container" onclick={handleClick}></lwc.dynamic>
+</template>
+```
+
+Imperatively:
+
+```javascript
+import { LightningElement } from "lwc";
+
+export default class DynamicCtor extends LightningElement {
+  customCtor;
+
+  connectedCallback() {
+    this.loadCtor();
+  }
+
+  async loadCtor() {
+    const { default: ctor } = await import("c/customConstructor");
+    this.customCtor = ctor;
+    this.template.querySelector(ctor.lightningElementName).setAttribute('title', 'hawaii');
+    this.template.querySelector(ctor.lightningElementName).addEventListener('click', () => console.log('hello world'));
```

Comment:
This will fail, the component will only be mounted in the DOM in the next rendering cycle.

Reply:
Fixed the [example](https://github.com/salesforce/lwc-rfcs/pull/71/files#diff-a1de05148a61ea9b00f70bd4ff5feb3895c16dd99707b173ba57c081b859250bR286-R301) with the latest commit.